Content.Server/Shipyard/Systems/ShipyardSystem.cs

```diff
@@ -0,0 +1,178 @@
+            _shipyardConsole.InitializeConsole();
+            SubscribeLocalEvent<ShipyardConsoleComponent, ComponentInit>(OnShipyardStartup);
+            SubscribeLocalEvent<RoundRestartCleanupEvent>(OnRoundRestart);
+        }
+
+        private void OnShipyardStartup(EntityUid uid, ShipyardConsoleComponent component, ComponentInit args)
+        {
+            if (!_enabled)
+                return;
+
+            SetupShipyard();
+        }
+
+        private void OnRoundRestart(RoundRestartCleanupEvent ev)
+        {
+            _configManager.UnsubValueChanged(CCVars.Shipyard, SetShipyardEnabled);
+            CleanupShipyard();
+        }
+
+        private void SetShipyardEnabled(bool value)
+        {
+            if (_enabled == value)
+                return;
+
+            _enabled = value;
+
+            if (value)
+            {
+                SetupShipyard();
+            }
+            else
+            {
+                CleanupShipyard();
+            }
+        }
+
+        /// <summary>
+        /// Adds a ship to the shipyard, calculates its price, and attempts to ftl-dock it to the given station
+        /// </summary>
+        /// <param name="stationUid">The ID of the station to dock the shuttle to</param>
+        /// <param name="shuttlePath">The path to the shuttle file to load. Must be a grid file!</param>
+        public void PurchaseShuttle(EntityUid? stationUid, string shuttlePath, out ShuttleComponent? vessel)
+        {
+            if (!TryComp<StationDataComponent>(stationUid, out var stationData) || !TryComp<ShuttleComponent>(AddShuttle(shuttlePath), out var shuttle))
+            {
+                vessel = null;
+                return;
+            }
+
+            var targetGrid = _station.GetLargestGrid(stationData);
+
+            if (targetGrid == null)
+            {
+                vessel = null;
+                return;
+            }
+
+            var price = _pricing.AppraiseGrid(shuttle.Owner, null);
+
+            //can do TryFTLDock later instead if we need to keep the shipyard map paused
+            _shuttle.FTLTravel(shuttle, targetGrid.Value, 0f, 30f, true);
+            vessel = shuttle;
+            _sawmill.Info($"Shuttle {shuttlePath} was purchased at {targetGrid} for {price}");
+        }
+
+        /// <summary>
+        /// Loads a shuttle into the ShipyardMap from a file path
+        /// </summary>
+        /// <param name="shuttlePath">The path to the grid file to load. Must be a grid file!</param>
+        /// <returns>Returns the EntityUid of the shuttle</returns>
+        private EntityUid? AddShuttle(string shuttlePath)
+        {
+            if (ShipyardMap == null)
+                return null;
+
+            var loadOptions = new MapLoadOptions()
+            {
+                Offset = (500f + _shuttleIndex, 0f)
+            };
+
+            if (!_map.TryLoad(ShipyardMap.Value, shuttlePath.ToString(), out var gridList, loadOptions) || gridList == null)
```

Comment:
```suggestion
            if (!_map.TryLoad(ShipyardMap.Value, shuttlePath, out var gridList, loadOptions))
```
Please stop with all the defensive null checks when it's compiler enforced.